File: Codigos estudiantes por lenguaje/C++/Cristian Maldonado/Matemática/Prime_Factors.cpp
#include <iostream>

using namespace std;

void primeFactors(int n){
    for(int p = 2; p * p <= n; p++){
        while(n % p == 0){
            printf("%d\n",p);
            n /= p;
        }
    }
    if(n > 1){
        printf("%d\n", n);
    }
}

int main()
{
    int n;
    scanf("%i", &n);
    primeFactors(n);
    return 0;
}

File: Codigos estudiantes por lenguaje/Java/Cristian Maldonado/Matemática/Prime_Factors.java

public class Prime_Factors {

    static void primeFactor(int n) {
        for (long p = 2; p * p <= n; p++) {
            while (n % p == 0) {
                System.out.println(p);
                n /= p;
            }
        }
        if (n > 1) {
            System.out.println(n);
        }
    }

    public static void main(String[] args) {
        primeFactor(20);
    }
}
